17: Routing II - TopBar

gabrielizalo opened the pull request:
# Summary

* 17: Routing II - TopBar
   Se adicionaron las siguientes páginas: Profile, Error 404, Offline.


# Trello Card

https://trello.com/c/uJVzXHFq/25-routing-ii-topbar


# Screenshot

![screenshot_2018-11-20-09-53-19](https://user-images.githubusercontent.com/656353/48782092-6bfb6380-ecab-11e8-84c5-d5cd97fa1d55.png)
![screenshot_2018-11-20-09-54-11](https://user-images.githubusercontent.com/656353/48782093-6bfb6380-ecab-11e8-87fc-7ee79c45276d.png)
![screenshot_2018-11-20-09-54-26](https://user-images.githubusercontent.com/656353/48782094-6bfb6380-ecab-11e8-9a0f-b527fe05a6cd.png)
<img width="770" alt="gabos tica tac toe - login" src="https://user-images.githubusercontent.com/656353/48782163-8d5c4f80-ecab-11e8-833a-73cfcb215a20.png">
<img width="770" alt="gabos tica tac toe - profile" src="https://user-images.githubusercontent.com/656353/48782164-8d5c4f80-ecab-11e8-98bb-3625a9f65800.png">
<img width="770" alt="gabos tica tac toe - game" src="https://user-images.githubusercontent.com/656353/48782166-8d5c4f80-ecab-11e8-8823-a707cfa35c95.png">
<img width="770" alt="gabos tica tac toe - 404" src="https://user-images.githubusercontent.com/656353/48782168-8d5c4f80-ecab-11e8-8dd6-124347d3f56b.png">
<img width="770" alt="gabos tica tac toe - offline" src="https://user-images.githubusercontent.com/656353/48782169-8d5c4f80-ecab-11e8-920a-82dddcd36631.png">

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Added the new component TopBar

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Changed the port of API Server to 3002
- Fixed that game page has only visible to logged users

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- The Top Bar is ready and it appears only with authenticated pages
- There is a new page for 'Profile'
- There is a new page for '404 Errors'
- Users can log out from the game

gabrielizalo committed:
Merge branch '16-Routing-I-Login-Part4' into 17-Routing-II-TopBar

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Added a page to use when we are offline

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Changed Offline setting to false

henryzarza commented on react/react-training/public/index.html:
```diff
@@ -17,6 +17,8 @@
     <meta name="msapplication-TileColor" content="#333132">
     <meta name="theme-color" content="#333132">
+
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
```
Move this to `application.scss`

henryzarza commented on react/react-training/src/app/components/AppRoutes/index.js:
```diff
@@ -2,21 +2,34 @@ import connect from 'react-redux/es/connect/connect';
+    // We are online...
+    let { userIsLogged } = this.props;
+    if (!userIsLogged) {
+      userIsLogged = localStorage.getItem('userIsLogged') === 'true';
```
Isn't a good practice modify the props

henryzarza commented on react/react-training/src/app/screens/Dashboard/index.js:
```diff
@@ -0,0 +1,26 @@
+function Dashboard() {
+  return (
+    <Router>
+      <Fragment>
```
This `Fragment` is required?

gabrielizalo replied:
@henryzarza yesss... Si no lo tengo me sale este error:
`A <Router> may have only one child element`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js:
```diff
@@ -0,0 +1,33 @@
+function TopBar() {
+  return (
+    <nav className={styles.bar}>
+      <ul className={styles.list}>
```
Isn't necessary this nesting? I recommend you modify the structure

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js:
```diff
@@ -0,0 +1,33 @@
+      <ul className={styles.list}>
+        <li className={styles.item}>
+          <i className="fas fa-gamepad" />
+          <NavLink activeClassName={styles.active} className={styles.link} to="/game/">
```
Put the routes as constants

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js:
```diff
@@ -0,0 +1,33 @@
+        </li>
+        <li className={styles.item}>
+          <i className="fas fa-user" />
+          <NavLink activeClassName={styles.active} className={styles.link} to="/profile/">
```
The same here

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js:
```diff
@@ -0,0 +1,33 @@
+        </li>
+        <li className={styles.item}>
+          <i className="fas fa-sign-out-alt" />
+          <NavLink activeClassName={styles.active} className={styles.link} to="/logout/">
```
And here

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/styles.scss:
```diff
@@ -0,0 +1,36 @@
+  padding: 15px;
+}
+
+.item:hover, .item a:hover{
```
Use the nesting of SASS

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Logout/index.js:
```diff
@@ -0,0 +1,28 @@
+  }
+
+  render() {
+    return <Fragment />;
```
I think that isn't necessary this screen, please delete it

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Logout/styles.scss:
```diff
@@ -0,0 +1,8 @@
+@import 'variables/colors';
```
And delete this

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/index.js:
```diff
@@ -0,0 +1,13 @@
+
+function Profile() {
+  return (
+    <div>
```
If you have only one element isn't necessary this

henryzarza commented on react/react-training/src/app/screens/Error404/index.js:
```diff
@@ -0,0 +1,19 @@
+    <div>
+      <h2 className={styles.title}>{ERROR_404}</h2>
+      <p className={styles.image}>
+        <img alt={ERROR_404} height={cry.height} src={cry} width={cry.width} />
```
Isn't a good practice to assign the properties `width` and `height` here, please move it to `styles` file

henryzarza commented on react/react-training/src/app/screens/Offline/index.js:
```diff
@@ -0,0 +1,19 @@
+    <div>
+      <h2 className={styles.title}>{OFFLINE_MESSAGE}</h2>
+      <p className={styles.image}>
+        <img alt={OFFLINE_MESSAGE} height={offline.height} src={offline} width={offline.width} />
```
The same here

henryzarza commented on react/react-training/src/scss/styles.scss:
```diff
@@ -2,7 +2,11 @@
   color: $white;
   font-family: $font-main;
+  justify-content: center;
+  margin: 0;
```
This is necessary?

henryzarza commented on react/react-training/src/scss/styles.scss:
```diff
@@ -2,7 +2,11 @@
   font-family: $font-main;
+  justify-content: center;
+  margin: 0;
+  padding: 0;
```
The same question here

henryzarza commented on react/react-training/src/scss/variables/_fonts.scss:
```diff
@@ -5,3 +5,7 @@ $font-main: 'Roboto', Helvetica, sans-serif;
 $button-font-size: 15px;
+
+$top-bar-size: 12px;
+
```
Delete this line

henryzarza commented on react/react-training/src/scss/variables/_fonts.scss:
```diff
@@ -5,3 +5,7 @@ $font-main: 'Roboto', Helvetica, sans-serif;
 $title-size: 25px;
 $boxes-font-size: 15px;
 $button-font-size: 15px;
+
```
Delete this line

wfolini commented on react/react-training/.env:
```diff
@@ -1 +1 @@
-REACT_APP_API_BASE_URL=http://localhost:3001/api/
+REACT_APP_API_BASE_URL=http://localhost:3002/api/
```
Remember to delete this file from repository

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Changed Offline setting to false

gabrielizalo committed:
Revert "Doing "ReactApp - Tic Tac Toe""

This reverts commit 1bda0de57a15f0d8abd804a93bb1753571220c2b.

gabrielizalo committed:
Merge remote-tracking branch 'origin/master' into 17-Routing-II-TopBar

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Moved external CSS Font Awesome to the styles.scss

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Fixed HTML of the Top Bar to a simpler one

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- In the Top Bar now the routes are the constants we already have

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Fixing styles, using the nesting of SASS

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Removed unnecessary div tag

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- The width and height of the images are now in the styles files

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Removed unnecessary styling

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Removed the unnecessary page: Logout
- Minor styling tweaks

gabrielizalo commented:
We are readyyyyyyy.. @WFolini make your move 👍

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Removed the file .env. It has sensitive information
- Updated README.md section

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Removed the file .env. It has sensitive information
- Updated README.md section

gabrielizalo committed:
Merge remote-tracking branch 'origin/17-Routing-II-TopBar' into 17-Routing-II-TopBar

gabrielizalo commented:
We are super ok 👍

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Updated README.md section
